Tests/test_AlignIO_FastaIO_unittest.py

```diff
@@ -0,0 +1,256 @@
+; al_stop: 81
+; al_display_start: 14
+DALLGEIQRLRKQVHQLQLERDILTKANELIKKDLGVSFLKLKNREKTLI
+VDALKKKYPVAELLSVLQLARSCYFYQNVCTISMRKYA
+  3>>>gi|10955265|ref|NP_052606.1| hypothetical protein pOSAK1_03 [Escherichia coli O157:H7 s 346 aa - 346 aa
+ vs  NC_009649.faa library
+  45119 residues in   180 sequences
+  Expectation_n fit: rho(ln(x))= 6.0276+/-0.0276; mu= 3.0670+/- 1.461
+ mean_var=37.1634+/- 8.980, 0's: 0 Z-trim: 1  B-trim: 14 in 1/25
+ Lambda= 0.210386
+FASTA (3.5 Sept 2006) function [optimized, BL50 matrix (15:-5)] ktup: 2
+ join: 37, opt: 25, open/ext: -10/-2, width:  16
+ Scan time:  0.020
+The best scores are:                                      opt bits E(180)
+gi|152973545|ref|YP_001338596.1| putative plasmid  ( 242)   70 27.5   0.082
+>>>gi|10955265|ref|NP_052606.1|, 346 aa vs NC_009649.faa library
+; pg_name: /opt/fasta/fasta34
+; pg_ver: 34.26
+; pg_argv: /opt/fasta/fasta34 -Q -H -E 1 -m 10 NC_002127.faa NC_009649.faa
+; pg_name: FASTA
+; pg_ver: 3.5 Sept 2006
+; pg_matrix: BL50 (15:-5)
+; pg_open-ext: -10 -2
+; pg_ktup: 2
+; pg_optcut: 25
+; pg_cgap: 37
+; mp_extrap: 60000 180
+; mp_stats:  Expectation_n fit: rho(ln(x))= 6.0276+/-0.0276; mu= 3.0670+/- 1.461  mean_var=37.1634+/- 8.980, 0's: 0 Z-trim: 1  B-trim: 14 in 1/25  Lambda= 0.210386
+; mp_KS: -0.0000 (N=0) at 8159228
+>>gi|152973545|ref|YP_001338596.1| putative plasmid SOS inhibition protein A [Klebsiella pneumoniae subsp. pneumoniae MGH 78578]
+; fa_frame: f
+; fa_initn:  52
+; fa_init1:  52
+; fa_opt:  70
+; fa_z-score: 105.5
+; fa_bits: 27.5
+; fa_expect:  0.082
+; sw_score: 70
+; sw_ident: 0.279
+; sw_sim: 0.651
+; sw_overlap: 43
+>gi|10955265| ..
+; sq_len: 346
+; sq_offset: 1
+; sq_type: p
+; al_start: 197
+; al_stop: 238
+; al_display_start: 167
+DFMCSILNMKEIVEQKNKEFNVDIKKETIESELHSKLPKSIDKIHEDIKK
+QLSC-SLIMKKIDVEMEDYSTYCFSALRAIEGFIYQILNDVCNPSSSKNL
+GEYFTENKPKYIIREIHQET
+>gi|152973545|ref|YP_001338596.1| ..
+; sq_len: 242
+; sq_type: p
+; al_start: 52
+; al_stop: 94
+; al_display_start: 22
+IMTVEEARQRGARLPSMPHVRTFLRLLTGCSRINSDVARRIPGIHRDPKD
+RLSSLKQVEEALDMLISSHGEYCPLPLTMDVQAENFPEVLHTRTVRRLKR
+QDFAFTRKMRREARQVEQSW
+>>><<<
+579 residues in 3 query   sequences
+45119 residues in 180 library sequences
+ Scomplib [34.26]
+ start: Tue May 20 16:38:45 2008 done: Tue May 20 16:38:45 2008
+ Total Scan time:  0.020 Total Display time:  0.010
+Function used was FASTA [version 34.26 January 12, 2007]
+"""
+
+
+class TestSelf(unittest.TestCase):
+    def test_example(self):
+        alignments = list(FastaIO.FastaM10Iterator(StringIO(simple_example)))
+        self.assertEqual(len(alignments), 4)
+        self.assertEqual(len(alignments[0]), 2)
+        rows = (2, 3, 4, 5)
+        cols = (108, 64, 123, 456)
+        for a, rows, cols in zip(alignments, rows, cols):
+            self.assertEqual(rows, len(a))
+            self.assertEqual(cols, a.get_alignment_length())
+            for r in a:
+                print("%s %s %i" % (r.seq, r.id, r.annotations["original_length"]))
```

Comment:
Rather than a print statement, this would ideally be set of comparisons:

``` python
self.assertEqual(r.seq, ...)
self.assertEqual(r.id, ...)
self.assertEqual(r.annotations["original_length"], ...)
```

You'd need to supply the expected values for each row of each of the four alignments, a bit like how the code above provides the expected number of rows and columns for each of the four alignments.